### ISSUE TITLE
Can't use 8bitdo SN30+ controller in-game

### ISSUE THREAD

**Issue by chena1** [mupen64plus-ae/mupen64plus-ae#756]
I bought an 8-bitdo SN30+ controller to play games on my Google Pixelbook. My controller connects via bluetooth. While I can create controller profiles and the button presses register in the Profiles>Controller> mapping menu, I can't use my controller in-game (I've tried with Banjo-Tooie, Diddy Kong Racing, and Pokemon Stadium-none of these games register when I press Start to skip the intro). 

I tried disabling the touchscreen options in the global settings, disabling Controllers 2 - 4 in the global settings and setting the Controller 1 profile to my custom profile. The issue persists. I noticed that in the Controller Diagnostics screen, my device shows up as Device 3, but I don't know how to change that. 

For the record, here is what I see in Hardware Info: 
Device: Pro Controller
Type: Default
Signature: 0,1,12,13,15,16
Hash: 278430807
  AXIS_X: Normal
  AXIS_Y: Normal
  AXIS_RX: Normal
  AXIS_RY: Normal
  AXIS_HAT_X: Normal
  AXIS_HAT_Y: Normal

Device: Nintendo Co., Ltd. Pro Controller
Type: Default
Signature: 0,1,11,14,15,16
Hash: 1362447993
  AXIS_X: Normal
  AXIS_Y: Normal
  AXIS_Z: Normal
  AXIS_RZ: Normal
  AXIS_HAT_X: Normal
  AXIS_HAT_Y: Normal

Device: ChromeOS Keyboard
Id: 2
Descriptor: 
Class: BUTTON

Device: Pro Controller
Id: 3
Descriptor: 
Class: BUTTON, JOYSTICK
Axes: 6
  AXIS_X (joystick): ( -1.0 , 1.0 )
  AXIS_Y (joystick): ( -1.0 , 1.0 )
  AXIS_RX (joystick): ( -1.0 , 1.0 )
  AXIS_RY (joystick): ( -1.0 , 1.0 )
  AXIS_HAT_X (joystick): ( -1.0 , 1.0 )
  AXIS_HAT_Y (joystick): ( -1.0 , 1.0 )

Device: Nintendo Co., Ltd. Pro Controller
Id: 4
Descriptor: 
Class: BUTTON, JOYSTICK
Axes: 6
  AXIS_X (joystick): ( -1.0 , 1.0 )
  AXIS_Y (joystick): ( -1.0 , 1.0 )
  AXIS_Z (joystick): ( -1.0 , 1.0 )
  AXIS_RZ (joystick): ( -1.0 , 1.0 )
  AXIS_HAT_X (joystick): ( -1.0 , 1.0 )
  AXIS_HAT_Y (joystick): ( -1.0 , 1.0 )

processor	: 0
vendor_id	: GenuineIntel
cpu family	: 6
model		: 142
model name	: Intel(R) Core(TM) i5-7Y57 CPU @ 1.20GHz
stepping	: 9
microcode	: 0xb4
cpu MHz		: 699.812
cache size	: 4096 KB
physical id	: 0
siblings	: 2
core id		: 0
cpu cores	: 2
apicid		: 0
initial apicid	: 0
fpu		: yes
fpu_exception	: yes
cpuid level	: 22
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch invpcid_single intel_pt ssbd ibrs ibpb stibp kaiser tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm mpx rdseed adx smap clflushopt xsaveopt xsavec xgetbv1 dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp md_clear flush_l1d
bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds
bogomips	: 3215.98
clflush size	: 64
cache_alignment	: 64
address sizes	: 39 bits physical, 48 bits virtual
power management:

processor	: 2
vendor_id	: GenuineIntel
cpu family	: 6
model		: 142
model name	: Intel(R) Core(TM) i5-7Y57 CPU @ 1.20GHz
stepping	: 9
microcode	: 0xb4
cpu MHz		: 700.000
cache size	: 4096 KB
physical id	: 0
siblings	: 2
core id		: 1
cpu cores	: 2
apicid		: 2
initial apicid	: 2
fpu		: yes
fpu_exception	: yes
cpuid level	: 22
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch invpcid_single intel_pt ssbd ibrs ibpb stibp kaiser tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm mpx rdseed adx smap clflushopt xsaveopt xsavec xgetbv1 dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp md_clear flush_l1d
bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds
bogomips	: 3215.98
clflush size	: 64
cache_alignment	: 64
address sizes	: 39 bits physical, 48 bits virtual
power management:

Board: eve
Brand: google
Device: eve_cheets
Display: R77-12371.75.3 release-keys
Host: wprf9.hot.corp.google.com
ID: R77-12371.75.3
Manufacturer: Google
Model: Google Pixelbook
Product: eve

**Comment by fzurita** [comment]
Go to profiles then select profiles.

In there, after selecting your profile for controller 1, there are a few more parameters you can play with.

To see if your controller will work at all, enable the "share controller" option and try starting the game and seeing if your controller works. If it does, I would then go back to the select profiles and select the "multiplayer" option. Manually map your Bluetooth controller to player 1 from there.

**Comment by chena1** [comment]
Thank you for your suggestions! I checked Share Controller in Global Settings and set Controller 2 in Global Settings to my custom option. I decided to test with Diddy Kong Racing. To be safe, I selected Yes to share controller in the game settings. I was able to get it responding (had to reset my custom mapping, idk why it seemed different) I noticed in the character select screen that my controller is listed as No 2 and not No 1. I mostly intend to play games solo, so that won’t bug me. Thanks again for your help!
